kernel/memory/src/lib.rs

```diff
@@ -140,7 +140,8 @@ pub fn set_broadcast_tlb_shootdown_cb(func: fn(PageRange)) {
 ///  7. the kernel's list of *other* higher-half MappedPages that needs to be converted to a vector after heap initialization, and which should be kept forever,
 ///  8. the kernel's list of identity-mapped MappedPages that needs to be converted to a vector after heap initialization, and which should be dropped before starting the first userspace program. 
 pub fn init(
-    boot_info: &impl BootInformation
+    boot_info: &impl BootInformation,
+    kernel_stack_start: VirtualAddress,
```

Comment:
can `kernel_stack_start` virt addr be added to `boot_info`? I think it makes sense but I may be missing something.

Reply:
Because the UEFI boot info does not contain the kernel stack start. The boot code immediately checks `rsp` and passes its value to `rust_entry`.